src/usr/local/buildpack/utils/node.sh

```diff
@@ -7,18 +7,49 @@ function npm_init () {
 }
 
 function npm_install () {
-  local tool_path=
-  tool_path="$(create_versioned_tool_path)"
+  local versioned_tool_path
+  versioned_tool_path="$(create_versioned_tool_path)"
 
-  npm install "${TOOL_NAME}@${TOOL_VERSION}" --global --no-audit --prefix "$tool_path" --cache "${NPM_CONFIG_CACHE}" 2>&1
+  # get semver => MAJOR, MINOR, PATCH
+  check_semver "${TOOL_VERSION}"
+
+  npm install "${TOOL_NAME}@${TOOL_VERSION}" --global --no-audit --prefix "$versioned_tool_path" --cache "${NPM_CONFIG_CACHE}" 2>&1
 
   if [[ "${TOOL_NAME}" == "npm" && ${MAJOR} -lt 7 ]]; then
     # update to latest node-gyp to fully support python3
-    "$tool_path/bin/npm" explore npm --global --prefix "$tool_path" -- npm install node-gyp@latest --no-audit --cache "${NPM_CONFIG_CACHE}" 2>&1
+    "$versioned_tool_path/bin/npm" explore npm --global --prefix "$versioned_tool_path" -- npm install node-gyp@latest --no-audit --cache "${NPM_CONFIG_CACHE}" 2>&1
   fi
 }
 
 function npm_clean () {
   # Clean npm stuff
   rm -rf "$HOME/.cache" "${NPM_CONFIG_CACHE}" "$HOME/.npm/_logs"/*
 }
+
+# Helper function to link to a globally installed node
```

Comment:
shouldn't those functions stay in node tool? as they are only used / allowed for node installer.